src/types/spot.ts

```diff
@@ -1929,3 +1924,3114 @@ export interface BSwapLiquidity {
+  address: string;
+  addressTag: string;
+  coin: string;
+  name: string;
+  network: string;
+  origin: string;
+  originType: string;
+  whiteStatus: boolean;
+}
+
+export interface AccountInfoResponse {
+  vipLevel: number;
+  isMarginEnabled: boolean;
+  isFutureEnabled: boolean;
+}
+
+export interface ManagedSubAccountSnapshotParams {
+  email: string;
+  type: 'SPOT' | 'MARGIN' | 'FUTURES';
+  startTime?: number;
+  endTime?: number;
+  limit?: number;
+}
+
+export interface ManagedSubAccountSnapshotResponse {
+  code: number;
+  msg: string;
+  snapshotVos: {
+    data: {
+      balances?: {
+        asset: string;
+        free: string;
+        locked: string;
+      }[];
+      totalAssetOfBtc?: string;
+      marginLevel?: string;
+      totalLiabilityOfBtc?: string;
+      totalNetAssetOfBtc?: string;
+      userAssets?: {
+        asset: string;
+        borrowed: string;
+        free: string;
+        interest: string;
+        locked: string;
+        netAsset: string;
+      }[];
+      assets?: {
+        asset: string;
+        marginBalance: string;
+        walletBalance: string;
+      }[];
+      position?: {
+        entryPrice: string;
+        markPrice: string;
+        positionAmt: string;
+        symbol: string;
+        unRealizedProfit: string;
+      }[];
+    };
+    type: string;
+    updateTime: number;
+  }[];
+}
+
+export interface ManagedSubAccountTransferLogParams {
+  email: string;
+  startTime: number;
+  endTime: number;
+  page: number;
+  limit: number;
+  transfers?: 'from' | 'to';
+  transferFunctionAccountType?:
+    | 'SPOT'
+    | 'MARGIN'
+    | 'ISOLATED_MARGIN'
+    | 'USDT_FUTURE'
+    | 'COIN_FUTURE';
+}
+
+export interface ManagedSubAccountTransferLogResponse {
+  managerSubTransferHistoryVos: {
+    fromEmail: string;
+    fromAccountType: string;
+    toEmail: string;
+    toAccountType: string;
+    asset: string;
+    amount: string;
+    scheduledData: number;
+    createTime: number;
+    status: string;
+    tranId: number;
+  }[];
+  count: number;
+}
+
+export interface ManagedSubAccountFuturesAssetsResponse {
+  code: string;
+  message: string;
+  snapshotVos: {
+    type: string;
+    updateTime: number;
+    data: {
+      assets: {
+        asset: string;
+        marginBalance: number;
+        walletBalance: number;
+      }[];
+      position: {
+        symbol: string;
+        entryPrice: number;
+        markPrice: number;
+        positionAmt: number;
+      }[];
+    };
+  }[];
+}
+
+export interface ManagedSubAccountMarginAssetsResponse {
+  marginLevel: string;
+  totalAssetOfBtc: string;
+  totalLiabilityOfBtc: string;
+  totalNetAssetOfBtc: string;
+  userAssets: {
+    asset: string;
+    borrowed: string;
+    free: string;
+    interest: string;
+    locked: string;
+    netAsset: string;
+  }[];
+}
+
+export interface SubAccountAssetsMasterResponse {
+  balances: {
+    asset: string;
+    free: string;
+    locked: string;
+  }[];
+}
```

Comment:
flatten this, so the part using it just has { balances: nameoftypehere[]; } instead. See the comment on getSubAccountAssetsMaster